score/parse.go

```diff
@@ -0,0 +1,40 @@
+package score
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+)
+
+// hiddenSecret is used to replace the global secret when parsing.
+var hiddenSecret = "hidden"
+
+// ErrScoreNotFound is returned if the parse string did not contain a
+// JSON score string.
+var ErrScoreNotFound = errors.New("score not found in string")
+
+// Parse returns a score object for the provided JSON string s
+// which contains secret.
+func Parse(s, secret string) (*Score, error) {
+	if strings.Contains(s, secret) {
+		var sc Score
+		err := json.Unmarshal([]byte(s), &sc)
+		if err == nil {
+			if sc.Secret == secret {
+				sc.Secret = hiddenSecret // overwrite secret
+			}
+			return &sc, nil
+		}
+		if strings.Contains(err.Error(), secret) {
+			// this is probably not necessary, but to be safe
+			return nil, errors.New("error suppressed to avoid revealing secret")
+		}
+		return nil, err
+	}
+	return nil, ErrScoreNotFound
+}
+
+// HasPrefix returns true if the provided string s has a parsable prefix string.
```

Comment:
If we have a manual text matching of the string we force everyone who make a score object to make sure it prints out the secret first, and after what I know, the order of fields in JSON can be in any order. This makes AG harder to use while writing test for teachers and adds to thing needed to be checked. 

I propose that we relay on the parsing tool to tell us if its parseble or not, or have a more general way to check for this, where we know it will detect it in any case possible. If the HasPrefix is needed, only searching for `{` will give a match for all possibilities of a score object.